projects/igniteui-angular/src/lib/icon/types.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+import { Theme } from 'igniteui-webcomponents/theming/types.js';
+
+// Exported internal types
+export type IconThemeKey = Theme | 'default';
+
+export type IconReferencePair = {
+  alias: IconMeta;
+  target: IconMeta;
+  overwrite: boolean;
+};
+export type IconType = "svg" | "font" | "liga";
+
+export type IconReference = IconMeta & FamilyMeta;
+
+export type MetaReference = {
+  alias: IconMeta;
+  target: Map<IconThemeKey, IconMeta>;
+};
+
+export interface FamilyMeta {
+    className: string;
+    type: IconType;
+    prefix?: string;
+}
+
+export interface IconFamily {
```

Comment:
And `defaultFamily` on the service makes this public API too. Unless we hide that property and consider it internal?